packages/core/src/NavigationContainer.tsx

```diff
@@ -242,6 +242,16 @@ const Container = React.forwardRef(function NavigationContainer(
   );
 
   React.useEffect(() => {
+    // eslint-disable-next-line no-undef
+    if (__DEV__) {
+      if (
+        typeof state !== 'undefined' &&
+        Object.prototype.toString.call(state) !== '[object Object]'
+      ) {
+        console.warn('Navigation state contains non-serialiable value');
```

Comment:
```suggestion
        console.warn("We found non-serializable values in the navigation state, which can break usage such as persisting and restoring state. This might happen if you passed a function or other non-serializable values in params. If you need to use functions in your options, you can use 'navigation.setOptions' instead.");
```